src/ui/FacetRange/FacetRange.ts

```diff
@@ -178,16 +164,78 @@ export class FacetRange extends Facet implements IComponentBindings {
 
   protected processNewGroupByResults(groupByResults: IGroupByResult) {
     if (groupByResults != null && this.options.ranges == null) {
-      groupByResults.values.sort((valueA, valueB) => {
-        const startEndA = valueA.value.split('..');
-        const startEndB = valueB.value.split('..');
-        if (this.options.dateField) {
-          return Date.parse(startEndA[0]) - Date.parse(startEndB[0]);
-        }
-        return Number(startEndA[0]) - Number(startEndB[0]);
-      });
+      groupByResults.values.sort((valueA, valueB) => this.sortRangeGroupByResults(valueA, valueB));
+      super.processNewGroupByResults(groupByResults);
+    }
+  }
+
+  private sortRangeGroupByResults(valueA: IGroupByValue, valueB: IGroupByValue) {
+    const startEndA = this.extractStartAndEndValue(valueA);
+    const startEndB = this.extractStartAndEndValue(valueB);
+    let firstValue: string;
+    let secondValue: string;
+
+    if (!startEndA && !startEndB) {
+      firstValue = valueA.value;
+      secondValue = valueB.value;
+    } else {
+      firstValue = startEndA.start;
+      secondValue = startEndB.start;
+    }
+
+    if (this.options.dateField) {
+      return Date.parse(firstValue) - Date.parse(secondValue);
+    }
+    return Number(firstValue) - Number(secondValue);
+  }
+
+  private translateValueCaptionFromFunctionName(facetValue: IIndexFieldValue) {
+    const { start, end } = this.extractStartAndEndValue(facetValue);
+    if (start == null || end == null) {
+      return null;
     }
-    super.processNewGroupByResults(groupByResults);
+
+    const helper = TemplateHelpers.getHelper(this.options.valueCaption);
+
+    if (helper != null) {
+      return `${helper.call(this, start)} - ${helper.call(this, end)}`;
+    } else {
+      const startConverted = start.match(/^[\+\-]?[0-9]+(\.[0-9]+)?$/) ? Number(start) : DateUtils.convertFromJsonDateIfNeeded(start);
```

Comment:
It would be nice to save the regex in a variable maybe called `dateFormat`.

Reply:
Regex objects are stateful. It would give incorrect results to reuse it in the same block scope.

https://stackoverflow.com/a/11477448

Reply:
Hmm, I did not know about this. That said, based on the SO link, it seems state is conserved between different calls when using `regex.exec(<string>)` but is not when using `<string>.match(regex)` as in this case?